src/connector/src/source/kafka/source/message.rs

```diff
@@ -39,22 +41,50 @@ impl KafkaMeta {
             .into()
     }
 
-    pub fn extract_headers(&self) -> Option<Datum> {
-        self.headers.as_ref().map(|headers| {
-            let header_item: Vec<Datum> = headers
+    pub fn extract_headers(
+        &self,
+        inner_field: Option<&str>,
+        data_type: Option<&PbDataType>,
+    ) -> Option<Datum> {
+        // return the exact header value if inner_field is specified, if not found, return None
+        if let Some(target_field) = inner_field {
```

Comment:
Perhaps you can just write two functions: `fn extract_headers()` and `fn extract_header_field(inner_field, data_type)`